### ISSUE TITLE
`valid-jsdoc`: Expand `matchDescription` to apply when certain fields are present, and to apply to fields which are description-like

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#11043]
**What rule do you want to change?**

`valid-jsdoc`

**Does this change cause the rule to produce more or fewer warnings?**

More warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

It could be a new option or an expansion in contexts where the `matchDescription` option applies (possibly allowing an array or object instead of a string so as to tweak the expected behavior).

**Please provide some example code that this change will affect:**

```js
/**
* @module myModuleWithNoDescription
*/
```

or

```js
/**
* @summary
* @example
*/
```

**What does the rule currently do for this code?**

Fails to report any issue.

**What will the rule do after it's changed?**

When a JSDoc code comment has certain tags present elsewhere in a block (and which is not one of those covered by the [function/class-based detection](https://github.com/eslint/eslint/blob/master/lib/rules/valid-jsdoc.js#L474-L486) of the current behavior), a description that fails to match the regular expression of `matchDescription` (or equivalent separate config) should also be able to apply.

These tags should, as I see it, be comprised of:
- Flagging an object of potentially broad purpose calling for a description `@module`, `@mixin`, `@namespace`, `@external`/`@host`
- Though more specialized in some ways, `@typedef`, `@event` objects may also be of enough importance for a description
- `@interface` if present should call for a description, particularly when within [virtual comments](http://usejsdoc.org/tags-interface.html#virtual-comments) where the lack of an associated function or class would necessitate its own warning
- While these may or may not be of general interest, they are generic enough that they may merit requiring a description as well: `@global`, `@kind`, and `@type`
- `@variation` and `@version`, though more concrete, typically are associated with documentation as well

Besides certain tags flagging the need for a description by their presence, certain other tags require a description and I think there should be an option for `matchDescription` to match *within* these tags.

- `@summary` and `@file`/`@fileoverview`/`@overview`, fields which are descriptive in content and broad in purpose
- While the current code checks classes, I'm not sure whether `@classdesc` can occur separately from a class, e.g., as its own separate comment, and if so, its text should also be queryable.
- Some additional fields, though fairly specialized, like `@copyright`, `@example`, `@see`, `@since`, `@deprecated`, `@todo` also have descriptive text areas which should never or rarely be empty; `@throws`, `@yields`/`@yield` and `@property` do likewise along with a possible type

Other tags may expect something additional and would be worthy of having a rule, but are less like "descriptions":
    - a concrete name (`@author` (with name and email), `@class` (optional), `@constructs` (optional)); `@event` or `@fires`/`@emits` or `@listens` (`<className>#[event:]<eventName>`), `@exports`, `@external`/`@host`, `@function` (optional), `@interface`, `@kind`, `@mixin`, `@requires`
    - a type `@enum` (optional), `@implements`; `@package` (for Closure only), `@private` (for Closure only), `@protected` (for Closure only); not `@public` or `@static` for Closure?; `@throws`, `@type`, `@typedef` (type and name path))
    - a type and name (`@constant`/`@const` (optional), `@member`/`@var`, `@module`, `@namespace`)
    - a name path (`@alias`, `@augments`/`@extends`, `@borrows` (namepath "as" namepath), `@callback` (optional), `@lends`, `@memberof`, `@mixes`, `@name`, `@see`, `@this`)
    - a value (`@default`/`@defaultvalue`)
    - a version or identifier (`@license`, `@since`, `@tutorial`, `@variation`, `@version`)
    - a status (`@access` (package|private|protected|public)
    - a link (`@link`, `@tutorial`)

...and thus seem less like "descriptions" as per the rule name under discussion).

And a few seem they might benefit from insisting on no description as they don't seem to allow it:

- `@abstract` ,`@async`, `@generator`, `@global`, `@hideconstructor`, `@ignore`, `@inheritdoc`, `@inner`, `@instance`, `@override`, `@readonly` (and see access types above referencing Closure under types above)
 
**Are you willing to submit a pull request to implement this change?**

Not at this time.

The above was based on experience expanding JSDoc to svg-edit, and feeling something missing with the inability to find broad purpose descriptions that were missing.

**Comment by nzakas** [comment]
Thanks for the suggestion. As of yesterday, we have decided to officially [end-of-life JSDoc support](https://eslint.org/blog/2018/11/jsdoc-end-of-life) in ESLint. All JSDoc features are now deprecated and we won't be fixing bugs or making any further improvements to those features.

We are recommending that people transition over to use the [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc) plugin instead of the built-in rules in ESLint.

Thanks for understanding and we apologize for any inconvenience.
